treeherder/services/pulse/consumers.py

```diff
@@ -162,6 +162,36 @@ def on_message(self, body, message):
         )
         message.ack()
 
+class PulsesConsumer(PulseConsumer):
+    queue_suffix = env("PULSE_SOMETHING", default="need")
+
+    def bindings(self):
+        rv = []
+        if self.source.get('hgmo'):
+            rv += HGMO_PUSH_BINDINGS
+        elif self.source.get('github'):
+            rv += GITHUB_PUSH_BINDINGS
+        else:
+            return TASKCLUSTER_TASK_BINDINGS 
+        return rv
+
+    @newrelic.agent.background_task(name='pulse-listener-tasks.on_message', group='Pulse Listener')
+    def on_message(self,body,message):
+        exchange = message.delivery_info['exchange']
+        routing_key = message.delivery_info['routing_key']
+        logger.debug('received job message from %s#%s', exchange, routing_key)
+        if exchange == 'exchange/taskcluster-queue/v1/.':
```

Comment:
Do exchanges end with `.`?

Reply:
This is method of using wildcard selector. Because ```exchange/taskcluster-queue/v1/``` ends with 3 type of messages. But all of them belong to the same class of Pulse_task